xds/internal/xdsclient/tests/cds_watchers_test.go

```diff
@@ -865,11 +964,30 @@ func (s) TestCDSWatch_PartialValid(t *testing.T) {
 // expected to wait for the watch timeout to expire before concluding that the
 // resource does not exist on the server
 func (s) TestCDSWatch_PartialResponse(t *testing.T) {
-	mgmtServer, nodeID, bootstrapContents, _, cleanup := e2e.SetupManagementServer(t, e2e.ManagementServerOptions{})
-	defer cleanup()
+	mgmtServer := e2e.StartManagementServer(t, e2e.ManagementServerOptions{})
+
+	nodeID := uuid.New().String()
+	bc, err := bootstrap.NewContentsForTesting(bootstrap.ConfigOptionsForTesting{
+		Servers: []json.RawMessage{[]byte(fmt.Sprintf(`{
+					"server_uri": %q,
+					"channel_creds": [{"type": "insecure"}]
+				}`, mgmtServer.Address))},
+		NodeID: nodeID,
+		Authorities: map[string]json.RawMessage{
+			// Xdstp style resource names used in this test use a url escaped
+			// version of t.Name as their authority, and the empty config
+			// results in the top-level xds server configuration being used for
+			// this authority.
+			url.PathEscape(t.Name()): []byte(`{}`),
```

Comment:
Similar to my earlier comment about PathEscape.

Reply:
Done.